src/test/scala/mesosphere/marathon/api/GroupApiServiceTest.scala

```diff
@@ -0,0 +1,97 @@
+package mesosphere.marathon
+package api
+
+import mesosphere.UnitTest
+import mesosphere.marathon.core.election.ElectionService
+import mesosphere.marathon.core.group.GroupManager
+import mesosphere.marathon.core.pod.{ BridgeNetwork, ContainerNetwork }
+import mesosphere.marathon.plugin.auth.Identity
+import mesosphere.marathon.raml.{ App, GroupUpdate, Network, NetworkMode }
+import mesosphere.marathon.state._
+import mesosphere.marathon.test.GroupCreation
+import mesosphere.marathon.state.PathId._
+
+import scala.collection.immutable.Seq
+import scala.concurrent.Future
+
+class GroupApiServiceTest extends UnitTest with GroupCreation {
+  implicit val identity: Identity = new Identity {}
+
+  "revert a version if version is provided" in {
+    Given("Group manager with the group version")
+    val groupManager = mock[GroupManager]
+    val groupId = PathId.empty
+    val version = Timestamp.now()
+    val groupWithOlderVersion = createGroup(groupId, version = version)
+    groupManager.group(groupId, version).returns(Future.successful(Some(groupWithOlderVersion)))
+    val f = Fixture(groupManager = groupManager)
+    When("Calling update with version provided")
+    val updatedGroup = f.groupApiService.updateGroup(createRootGroup(), PathId.empty, GroupUpdate(version = Some(version.toOffsetDateTime)), version).futureValue
+
+    Then("Group of the provided version will be returned")
+    updatedGroup.group(groupId) should be (Some(groupWithOlderVersion))
+  }
+
+  "reverting to non-existing version throws exception" in {
+    Given("Group manager with no group of required version")
+    val groupManager = mock[GroupManager]
+    val groupId = PathId.empty
+    val version = Timestamp.now
+    groupManager.group(groupId, version).returns(Future.successful(None))
+    val f = Fixture(groupManager = groupManager)
+
+    When("Calling update with version provided")
+    Then("Exception will be thrown")
+    val ex = f.groupApiService.updateGroup(createRootGroup(), PathId.empty, GroupUpdate(version = Some(version.toOffsetDateTime)), version).failed.futureValue
```

Comment:
You could also do

```scala
an[IllegalArgumentException] should be thrownBy {
  f.groupApiService
   .updateGroup(createRootGroup(), 
                PathId.empty, 
                GroupUpdate(version = Some(version.toOffsetDateTime)),
                version)
   .futureValue
}
```

Reply:
unfortunately I can't because of https://github.com/scalatest/scalatest/issues/1172

Reply:
Ah, bummer.